added ReferenceCount and ReferenceByIndex functions to script adapter

fireundubh opened the pull request:
- added ReferenceCount and ReferenceByIndex functions to script adapter
- renamed Reference properties for consistency with ReferencedBy properties (per discussion following the implementation of these properties)

fireundubh committed:
renamed Reference properties for consistency with ReferencedBy properties

fireundubh committed:
added ReferenceCount and ReferenceByIndex functions to script adapter

ElminsterAU commented on Core/wbImplementation.pas:
```diff
@@ -1181,9 +1181,9 @@   TwbMainRecord = class(TwbRecord, IwbMainRecord, IwbMainRecordInternal, IwbMain
-    function GetReferences(aIndex: Integer): IwbMainRecord;
-    function GetReferencesCount: Integer;
-    function GetExternalReferencesCount: Integer;
+    function GetReferenceBy(aIndex: Integer): IwbMainRecord;
```
function GetReference(aIndex: Integer): IwbMainRecord;

ElminsterAU commented on Core/wbInterface.pas:
```diff
@@ -1726,12 +1726,12 @@   EwbSkipLoad = class(EAbort);
-      read GetReferencesCount;
-    property ExternalReferencedsCount: Integer
-      read GetExternalReferencesCount;
+    property ReferenceBy[aIndex: Integer]: IwbMainRecord
```
property References[aIndex: Integer]: IwbMainRecord

ElminsterAU commented on Core/wbImplementation.pas:
```diff
@@ -1181,9 +1181,9 @@   TwbMainRecord = class(TwbRecord, IwbMainRecord, IwbMainRecordInternal, IwbMain
-    function GetReferencesCount: Integer;
-    function GetExternalReferencesCount: Integer;
+    function GetReferenceBy(aIndex: Integer): IwbMainRecord;
+    function GetReferenceCount: Integer;
```
function GetReferencesCount: Integer;

ElminsterAU commented on Core/wbImplementation.pas:
```diff
@@ -1181,9 +1181,9 @@   TwbMainRecord = class(TwbRecord, IwbMainRecord, IwbMainRecordInternal, IwbMain
-    function GetExternalReferencesCount: Integer;
+    function GetReferenceBy(aIndex: Integer): IwbMainRecord;
+    function GetReferenceCount: Integer;
+    function GetExternalReferenceCount: Integer;
```
function GetExternalReferencesCount: Integer;

ElminsterAU commented on Core/wbInterface.pas:
```diff
@@ -1726,12 +1726,12 @@   EwbSkipLoad = class(EAbort);
-      read GetExternalReferencesCount;
+    property ReferenceBy[aIndex: Integer]: IwbMainRecord
+      read GetReferenceBy;
+    property ReferenceCount: Integer
```
property ReferencesCount: Integer

ElminsterAU commented on Core/wbInterface.pas:
```diff
@@ -1726,12 +1726,12 @@   EwbSkipLoad = class(EAbort);
+      read GetReferenceBy;
+    property ReferenceCount: Integer
+      read GetReferenceCount;
+    property ExternalReferenceCount: Integer
```
property ExternalReferencesCount: Integer

fireundubh committed:
renamed GetReferenceBy to GetReference

fireundubh committed:
renamed GetReferenceCount to GetReferencesCount
renamed GetExternalReferenceCount to GetExternalReferencesCount
renamed ReferencedBy property to References

fireundubh committed:
renamed ReferenceCount script function to ReferencesCount
renamed ReferenceByIndex script function to ReferencesByIndex

fireundubh commented:
Okay, I think I resolved all your requested changes. Please verify.
